distribution/pull_v2.go

```diff
@@ -1054,8 +1063,13 @@ func createDownloadFile() (*os.File, error) {
 	return os.CreateTemp("", "GetImageBlob")
 }
 
-func toOCIPlatform(p manifestlist.PlatformSpec) specs.Platform {
-	return specs.Platform{
+func toOCIPlatform(p manifestlist.PlatformSpec) *specs.Platform {
+	// distribution pkg does define platform as pointer so this hack for empty struct
+	// is necessary. This is temporary until correct OCI image-spec packge is used.
+	if p.OS == "" && p.Architecture == "" && p.Variant == "" && p.OSVersion == "" && p.OSFeatures == nil && p.Features == nil {
```

Comment:
Looks like the pinter picked up a typo;

```
 distribution/pull_v2.go:1068:66: `packge` is a misspelling of `package` (misspell)
	// is necessary. This is temporary until correct OCI image-spec packge is used.
 	                                                                ^
```


```suggestion
	// is necessary. This is temporary until correct OCI image-spec package is used.
	if p.OS == "" && p.Architecture == "" && p.Variant == "" && p.OSVersion == "" && p.OSFeatures == nil && p.Features == nil {
```